### PR TITLE
Rename support

### PR DESCRIPTION
**Pull request by Prince781**
Implement the `textDocument/rename` method of the protocol. Other changes are made to VLS to improve analysis of references to symbols.

### CODE REVIEW

**Comment by Prince781** [comment]
I've covered every edge case I can think of now, so I will wait a few days and if I can't think of anything then this will be merged.

**Comment by Prince781** [comment]
@benwaffle do you want to take a quick glance over this before I merge it?

File: src/codehelp/symbolvisitor.vala [hunk 1]
```diff
@@ -0,0 +1,732 @@
...
+        this.file = file;
+        this.symbol = symbol;
+        this.data = data;
+        this.seen = new Gee.HashSet<CodeNode> ();
+        this.include_declaration = include_declaration;
+        this.func = func;
+        visit_source_file (file);
+
+        // XXX: sometimes the CodeVisitor does not see a local variable,
+        // especially if it is declared as part of a foreach statement
```
Comment (benwaffle) [inline src/codehelp/symbolvisitor.vala:31]: sounds like a missing `accept_children` somewhere in valac

**Comment by Prince781** [comment]
Alright, I think I'm going to merge this now.

**Pull request merged**